ector/src/actor.rs

```diff
@@ -103,21 +85,74 @@ where
     }
 }
 
-impl<'a, M, R, MUT, const N: usize> ActorRequest<M, R> for Sender<'a, MUT, Request<M, R>, N>
+/// A trait send message to another actor and receives a reply.
+///
+/// Use [RequestManager] for typical usage.
+/// Note that the actor fufilling this request should always reply.
+/// The request can be cancelled and the uncompleted request should be cancelled internally
+pub trait ActorRequest<M, R> {
+    /// Attempts to send a message and wait for the response
+    /// Takes a mutable reference to avoid making multiple request from the same instnace
+    async fn request(&mut self, message: M) -> R;
+}
+
+pub struct RequestManager<A, M, R, MUT = NoopRawMutex>
 where
-    M: 'static,
+    A: ActorAddress<Request<M, R>>,
     MUT: RawMutex + 'static,
+    M: 'static,
+    R: 'static,
 {
-    async fn request(&self, message: M) -> R {
-        let channel: Channel<MUT, R, 1> = Channel::new();
-        let sender: DynamicSender<'_, R> = channel.sender().into();
+    addr: A,
+    reply_from: Sender<'static, MUT, R, 1>,
+    reply_to: Receiver<'static, MUT, R, 1>,
+    cancelled: usize,
+    _phatom: PhantomData<M>,
+}
+
+impl<A, M, R, MUT> RequestManager<A, M, R, MUT>
+where
+    A: ActorAddress<Request<M, R>>,
+    MUT: RawMutex,
+{
+    /// Creates a new instance of the request manager.
+    /// The channel must be provided by the user and only used by this manager
+    pub fn new(addr: A, channel: &'static mut Channel<MUT, R, 1>) -> Self
+    where
+        MUT: RawMutex,
+    {
+        Self {
+            addr,
+            reply_from: channel.sender(),
+            reply_to: channel.receiver(),
+            cancelled: 0,
+            _phatom: PhantomData,
+        }
+    }
+
+    async fn handle_missed(&mut self) {
+        while self.cancelled > 0 {
+            self.reply_to.recv().await;
```

Comment:
Would it be better to attempt draining the channel using try_recv()? Or do we always know that the other side will send the reply?

Reply:
Good question, didn't think about draining the channel. Thing is we might not receive the response associated with the right request. 

- Sender: Sends a request 1
- Actor: Starts handling request 1
- Sender: Cancels request 1
- Sender: Sends request 2 (drains channel, but nothing in it yet)
- Actor: Completes request 1
- Sender: Receives response from request 1 instead of 2

We can either
- Always assume a response is sent back
- Or assume that is not the case and accept that we might receive the response from the previous request when doing cancellation
- Or have a way to let the `Request` struct know that request was cancelled and no send the response when the request was cancelled. Would probably be ideal but I don't know yet how we could implement this.